File: drive_java/src/main/java/com/github/ximik3/gradle/ApkUpgradeType.java
package com.github.ximik3.gradle;

/**
 * 用來設定版本更新形式(依 minVersion) 1: minVersion 不變 2: minVersion 設為上一版 version 3: 強制更新
 **/
enum ApkUpgradeType {
    MIN_VERSION_NOT_CHANGE(1),
    MIN_VERSION_IS_CURRENT_ONLINE_VERSION(2),
    FORCE_UPGRADE(3),
    UNDEFINED(0);

    int value;

    ApkUpgradeType(int value) {
        this.value = value;
    }

    public static ApkUpgradeType parse(int value) {
        switch (value) {
            case 1:
                return MIN_VERSION_NOT_CHANGE;
            case 2:
                return MIN_VERSION_IS_CURRENT_ONLINE_VERSION;
            case 3:
                return FORCE_UPGRADE;
            default:
                return UNDEFINED;
        }
    }
}

File: drive_kotlin/src/main/kotlin/com/github/ximik3/gradle/enums/ApkUpgradeType.kt
package com.github.ximik3.gradle.enums

/**
 * 用來設定版本更新形式(依 minVersion)
 * 1: 不強更, minVersion 不變
 * 2: 強更至上一版, minVersion 設為上一版 version
 * 3: 強更至最新版
 */
enum class ApkUpgradeType(var value: Int) {
    NOT_FORCE_UPGRADE(1),
    FORCE_UPGRADE_TO_LAST_VERSION(2),
    FORCE_UPGRADE_TO_NEW_VERSION(3),
    UNDEFINED(0);

    companion object {
        fun parse(value: Int): ApkUpgradeType {
            return when (value) {
                1 -> NOT_FORCE_UPGRADE
                2 -> FORCE_UPGRADE_TO_LAST_VERSION
                3 -> FORCE_UPGRADE_TO_NEW_VERSION
                else -> UNDEFINED
            }
        }
    }
}